modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/refresh/RefreshPodcastsTask.kt

```diff
@@ -59,21 +60,36 @@ class RefreshPodcastsTask @AssistedInject constructor(
                 workManager.cancelAllWorkByTag(TAG_REFRESH_TASK)
                 return
             }
+            val request = buildPeriodicWorkRequest(settings)
 
-            val constraints = Constraints.Builder()
-                .setRequiredNetworkType(NetworkType.CONNECTED)
-                .setRequiresBatteryNotLow(true)
-                .build()
+            workManager.enqueueUniquePeriodicWork(TAG_REFRESH_TASK, ExistingPeriodicWorkPolicy.REPLACE, request)
+
+            LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "Set up periodic refresh")
+        }
+
+        private fun buildPeriodicWorkRequest(settings: Settings): PeriodicWorkRequest {
+            val constraints = buildPeriodicWorkRequestConstraints(settings)
 
-            val request = PeriodicWorkRequestBuilder<RefreshPodcastsTask>(REFRESH_EVERY_HOURS, TimeUnit.HOURS)
+            return PeriodicWorkRequestBuilder<RefreshPodcastsTask>(REFRESH_EVERY_HOURS, TimeUnit.HOURS)
                 .addTag(TAG_REFRESH_TASK)
                 .setConstraints(constraints)
                 .setInitialDelay(REFRESH_EVERY_HOURS, TimeUnit.HOURS)
                 .build()
+        }
 
-            workManager.enqueueUniquePeriodicWork(TAG_REFRESH_TASK, ExistingPeriodicWorkPolicy.REPLACE, request)
+        private fun buildPeriodicWorkRequestConstraints(settings: Settings): Constraints {
```

Comment:
Moved the constraint building out of the `scheduleOrCancel` method so it could be tested if desired.